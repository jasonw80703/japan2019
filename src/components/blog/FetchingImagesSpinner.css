.loading-spinner-div {
  margin: 0 auto;
  padding: 5%;
  text-align: center;
  border: 1px solid black;
  width: 50%;
}

.loading-spinner {
  width: 50px;
  height: 50px;
  margin: 0 auto;
}

.loading-text {
  margin-top: 1%;
}