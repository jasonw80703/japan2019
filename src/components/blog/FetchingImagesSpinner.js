import React from 'react';
import Spinner from 'react-bootstrap/Spinner'
import './FetchingImagesSpinner.css';

export default function FetchingImagesSpinner() {
  return (
    <div className="loading-spinner-div">
      <Spinner animation="border" className="loading-spinner" />
      <h2 className="loading-text">Gathering Images...</h2>
    </div>
  );
}