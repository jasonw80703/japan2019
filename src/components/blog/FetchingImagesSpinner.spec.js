import React from 'react';
import FetchingImagesSpinner from './FetchingImagesSpinner';
import renderer from 'react-test-renderer';

describe('<FetchingImagesSpinner />', () => {
  beforeEach(() => {
    jest.spyOn(global.Math, 'random').mockReturnValue(0.5);
  });

  afterEach(() => {
    jest.spyOn(global.Math, 'random').mockRestore();
  });

  it('should render as expected', () => {
    const tree = renderer
      .create(<FetchingImagesSpinner />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});