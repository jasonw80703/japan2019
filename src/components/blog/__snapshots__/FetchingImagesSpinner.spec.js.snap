// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FetchingImagesSpinner /> should render as expected 1`] = `
<div
  className="loading-spinner-div"
>
  <div
    className="loading-spinner spinner-border"
  />
  <h2
    className="loading-text"
  >
    Fetching Images...
  </h2>
  <p>
    <b>
      Japan Fun Fact:
    </b>
     
    Tokyo is the second most expensive city in the world to live in.
  </p>
</div>
`;
